src/merkle.rs

```diff
@@ -2,8 +2,15 @@ use sha3::{Digest, Sha3_256};
 
 type Hash = [u8; 32];
 
-struct Merkle {
-    tree: Vec<Hash>, // This will be a binary tree represented as a vector
+#[derive(Debug)]
+struct MerkleTree {
+    tree: Vec<Vec<Hash>>, // This will be a binary tree represented as a vector, with each level as a sub-vector with the hashes
+    leaves: Vec<Hash>,    // This will be uses to add new elements to the tree
+}
+
+#[derive(Debug)]
+struct MerkleProof {
+    proof: Vec<Hash>,
 }
```

Comment:
A struct might be unnecessary for this, for now just using Vec<Hash> should be enough

Reply:
Done